src/http/server/handlers/log_handler.cr

```diff
@@ -10,7 +10,9 @@ class HTTP::LogHandler
 
     Log.info { "#{context.request.method} #{context.request.resource} - #{context.response.status_code} (#{elapsed_text})" }
   rescue e
-    Log.error(exception: e) { "#{context.request.method} #{context.request.resource} - Unhandled exception:" }
+    unless context.response.closed?
```

Comment:
What's the reason for this? I'd expect an error to show up in the logs, even if the io is closed. The response could even have been sent!

Reply:
I was just thinking the same because even though the client might have closed the connection there was still some cpu being burned, so something must be logged. We could actually log just as a warning (or notice if we accept #9107) because there is nothing doable to fix the problem.
Would you still log the exception in that case?

Reply:
As long as it is an app error, it should definitely be logged. The IO might have actually been closed after the exception was raised.

Reply:
Under normal circumstances the output shouldn't be closed at that point.

The intention is to avoid logging as error the `IO::Error` caused by a client disconnection. Thus at `Response` when there is an error writing or flushing to the output it just mark the `Response::Output` as closed and re-raise the exception.

Maybe I should take a different approach, because the original `IO::Error` might have been handled and the exception rescued here is a different one. I'm thinking about wrapping the `IO::Error` within a more distinguishable exception here, like `ClientError`. So, unless the handler code doesn't rescue that exception (if only there was a way to make some exceptions not so easy to catch 😛), it will bubble up but we know for sure it doesn't have to be logged as error. It would still mark the output as closed so we don't keep trying to write and get the same error again. The only drawback I see is that raising a new exception makes it unwind the stack again.

I'll try to work around that idea and improve this.

Reply:
I think `IO::Error` should contain a reference to the `IO` which errored, then you can compare against the request and response IOs, and you know whether the `IO::Error` was related to the client or to another unrelated error.

Alternatively, we can just try not to be too smart and stop hiding client errors.

Reply:
Using the reference to the `IO` would break when the `IO` is actually wrapped. We would be forced to handle all errors and re-raise them updating the value.

Regarding hiding client errors, some days ago you agreed it was right to not show client disconnections as errors 🙂: https://github.com/crystal-lang/crystal/pull/9034#discussion_r406977081

I think this is (in part) what is making people complain about "broken pipe" errors in the logs all the time.

Reply:
Now the concerns related with closed IO and client errors are separated